### DISCUSSION TITLE
Vibrational spectra prediction

### DISCUSSION THREAD

**Discussion by nitbha007** [ACEsuit/mace#247] [Q&A] [answered]
Hello,
I want to use the "EnergyDipolesMACE" MACE model to predict the vibrational spectra. Can you direct me to the script which you used to obtain the spectra in your paper? Are you using auto-correlation function path or taking the derivative of the dipole moment to have the spectra.

Thank you!

**Comment by davkovacs** [comment]
Hi! Can you say please which spectra you are talking about?

**Reply by nitbha007** [reply]
Hi, IR and Raman spectra.

**Reply by davkovacs** [reply]
But you are looking for a script which I used to create a spectra in a paper? What paper and spectra do you mean :D

**Reply by nitbha007** [reply]
This paper: https://doi.org/10.1063/5.0155322

**Reply by davkovacs** [reply, marked as answer]
In that paper there is no IR or Raman spectra. I just computed the Fourier-transform of the velocity-velocity autocorrelation function. 

If you want IR and Raman I would recommend just using a simple MACE or ScaleShiftMACE to generate the trajectory and use a separate `AtomicDipolesMACE` to compute the dipoles and polarizabilities after the trajectory is generated. 

Please use this PR for trainign the dipole prediction model (I will try to merge it after Christmas ) https://github.com/ACEsuit/mace/pull/149

And here is an example script for fitting the model for the dipoles and polarisabilities:

```sh
python /PATH_TO_MACE/mace/scripts/run_train.py \
    --name="MACE_mu_alpha" \
    --train_file="train.xyz" \
    --valid_fraction=0.05 \
    --test_file="test.xyz" \
    --model="AtomicDipolesMACE" \
    --num_channels=32 \
    --max_L=2 \
    --r_max=5.0 \
    --loss="dipole" \
    --dipole_weight=100.0 \
    --polarizability_weight=100.0 \
    --dipole_key="mu" \
    --polarizability_key="alpha" \
    --batch_size=2 \
    --valid_batch_size=2 \
    --max_num_epochs=75 \
    --scheduler_patience=6 \
    --patience=12 \
    --ema \
    --error_table='DipoleRMSE' \
    --default_dtype="float64"\
    --device=cuda \
    --seed=123 \
    --restart_latest \
    --save_cpu \
```

And to evaluate the trajectory youu can use something like this:

```sh
python /PATH_TO_MACE/mace/scripts/eval_mu_alpha.py \
    --configs="md_simulation.xyz" \
    --model="/PATH_TO_MODEL/MACE_mu_alpha.model" \
    --output="mu_alpha_of_MD.xyz" \
    --device=cuda \
    --batch_size=128 \
```

Please make sure to adjust your batch size depending on the size of the training set for training and the size of the cells / molecules for testing.

**Reply by nitbha007** [reply]
Thank you. I will use the recommended PR!

**Reply by nitbha007** [reply]
Hello @davkovacs, Is there a quick way to log the error table at the end of model training? I saw that the logging information is commented out for writing the error about training, validation and test in the run_train.py script in this PR. 

Thank you!

**Reply by davkovacs** [reply]
I probably didn't have time to implement that. Feel free to add it and submit PR.

**Comment by mahrossi** [comment]
hey @davkovacs - just in case I want to use this branch/PR to predict only the polarizability (no dipoles) do you foresee any issues? I will try hands-on tonight or tomorrow, but just in case you see this and/or have any tips, I would be super grateful.

**Reply by mahrossi** [reply]
ok I can't seem to easily fool the code. this is a situation where calculating the dipoles are actually more complicated than calculating the polarizabilities (I know, crazy, but periodic systems). if you have a bit of time I would appreciate a quick chat - I think changing completely should be easy, but I got stuck in some routines in `mace/modules` where I am less sure about what the code is doing.

**Reply by ilyes319** [reply]
hey @mahrossi, if I understand correctly you want a method that computes only the polarizability ? You can easily just use the "AtomicDipolesMACE" model and put the loss of the dipole to zero with `--dipole_weight=0.0` in the input script. The model will automatically ignore that.

**Reply by mahrossi** [reply]
ok, I did that - I just also wanted to get some diagonstics directly on the polarizability MAEs and RMSEs -- but the way it is currently implemented that comes only as a byproduct of the dipole stuff (which would be meaningless in this case). So I just changed that to get me only polarizability related quantities. It worked already, all is trained ;-)

**Reply by mahrossi** [reply]
(the loss flattened at some value which gives larg-ish errors though (~12% or less in the individual tensor components) - probably some problem with my input, which I will chase

**Reply by ilyes319** [reply]
What unit do you currently use? It could have a numerical impact.

**Reply by mahrossi** [reply]
I am using bohr^3 for the polarizabilities (so, atomic units). any suggestions?

**Reply by ilyes319** [reply]
Could you try Ang^3? It might help with normalization (not sure).

**Reply by mahrossi** [reply]
yes I'll try it. I noticed that some scalings and so on are only done on the dipoles (which I am not using here). can that be a problem? sorry, I just read the code quickly, so I might have gotten the wrong impression.
